source/lib/predicates/number.ts

```diff
@@ -131,4 +131,11 @@ export class NumberPredicate extends Predicate<number> {
 			validator: value => value < 0
 		});
 	}
+
+	get integerOrInfinity() {
+		return this.addValidator({
+			message: (value, label) => `Expected ${label} to be integer or infinity, got ${value}`,
+			validator: value => value > 0 && (is.integer(value) || value === Infinity)
```

Comment:
The name `integerOrInfinity` does not reflect the behavior if positive, non-zero numbers are to be enforced.